frontend/src/metabase-lib/lib/DimensionOptions/DimensionOptions.ts

```diff
@@ -0,0 +1,69 @@
+import Dimension from "metabase-lib/lib/Dimension";
+import {
+  IDimensionFK,
+  IDimensionOptions,
+  IDimensionOptionsProps,
+  ISection,
+} from "./types";
+
+export default class DimensionOptions implements IDimensionOptions {
```

Comment:
We actually don't use the inheritance pattern at all (and `I` prefix for the interface).

Reply:
> We actually don't use the inheritance pattern at all

Since we are moving towards TypeScript, is it a problem to start using it?

Reply:
Probably warrants an RFC to initiate the discussion?

If you look at the few refactoring that I've done to this directory, the idea is to move away from half-baked OOP-ism towards data-oriented and immutability.

Reply:
> the idea is to move away from half-baked OOP-ism towards data-oriented and immutability

💯 but trying to move to TypeScript only atm and touching the logic the less possible.

Reply:
> but trying to move to TypeScript only atm and touching the logic the less possible.

Fair enough!